### DISCUSSION TITLE
Is this still valid? CADDY_GLOBAL_OPTIONS="servers { enable_full_duplex }"

### DISCUSSION THREAD

**Discussion by skeemer** [php/frankenphp#1088] [Q&A] [answered]
I'm trying to set protocols h2c via an environment variable. On https://frankenphp.dev/docs/config/#full-duplex-http1 it looked like I found my solution so I could do this.
```env
CADDY_GLOBAL_OPTIONS="servers { protocols h2c }"
```
This resulted in an error, so I tried the original from the docs to just see if it works and got the same error.
```env
CADDY_GLOBAL_OPTIONS="servers { enable_full_duplex }"
```

```log
Error: adapting config using caddyfile: Unexpected next token after '{' on same line, at /etc/caddy/Caddyfile:2
```

Are the docs correct?
The only thing I could find is https://github.com/klzgrad/naiveproxy/issues/323 which seems to indicate that you can't do this type of thing on a single line anymore.

**Comment by withinboredom** [comment, marked as answer]
full duplex only applies to http 1.1, not http2.

As for your error, that issue is indeed correct: you need to have line endings in there.

**Comment by skeemer** [comment]
Ok, so the docs are wrong.

I don't care about full duplex, I just tried that since that's what the docs had, so I thought it should work.

**Reply by dunglas** [reply]
Indeed. Would you mind open in a PR to fix the docs?

**Reply by skeemer** [reply]
Took me a while, but here you go: #1290
